src/Home/index.spec.tsx

```diff
@@ -1,16 +0,0 @@
-import React from "react"
-import { render } from "@testing-library/react-native"
-import { useFocusEffect } from "@react-navigation/native"
-
-import HomeScreen from "./index"
-
-jest.mock("@react-navigation/native")
-;(useFocusEffect as jest.Mock).mockReturnValue(jest.fn())
-
-describe("HomeScreen", () => {
-  it("renders", () => {
```

Comment:
This is a matter of how we organize these components, and how we test them afterward. 
- Should we be testing this one with all the setup? Context and all but just the happy path and then the presentational one with all the bells and whistles and all the branches of execution? 
- Should we not be testing this one at all?

Reply:
I think this one makes sense to be able to test the branches as they are quite important.

We will want some better coverage around, showing the possible states with ctas of 
```
// ios
1. en are authorized and app is enabled
2. en are authorized and app is not enabled
3. en is not authorized

// android
1. en are enabled
2. en are not enabled
```

Reply:
so i would tend to taking a pass at some TDD on this and see where the component wants to go as far as split.

Reply:
theres also tests we want to have around the listening for updates to the auth status via events

Reply:
I'm not sure I'm following here. This means that we should not test the presentational component in isolation? Like we do on the `Home.spec.tsx`? So should I move that test one level up and just test this entry file with all the branches and callback logic and then remove/rename the component test?

Reply:
I believe my concern is more about how we are organizing some of these components that receive some data from the context on an index file and then render and make decisions on the data on the more visually packed component. Is this in an effort to make testing easier by not having to have a context provider to test those or is it to separate logic from view?

Reply:
> Is this in an effort to make testing easier by not having to have a context provider to test those or is it to separate logic from view?

Imo, these are both positives gained by having them separate. In general i like things that tend towards functional core / imperative shell.

As far as testing the index/thing that connects to the provider/does IO: its hard to test IO stuff in general so keeping IO co located and at the boundary of the system helps. We may need a fake or some pretty heavy mocking to make a useful spec suite.

That all said, i definitely defer to you/whoever is writing the specs to what makes sense in this case in particular.  If it turns out better to consolidate the components we should do that.